### PR TITLE
fixed mac os build

### LINKED ISSUE #7: macOS 11.5.2 build fails because of OSSpinLock deprecation
**Describe the bug**

**To Reproduce**
Build using `make mac_osx` on Big Sur 11.5.2

**Expected behavior**
A clean build

**Hardware and operating system**
CPU: 2 GHz Quad-Core Intel Core i7
Memory: 8 GB 1600 MHz DDR3
Form factor: laptop
CPU arch: x86
Mac OS X: 11.5.2

**Additional context**

```
In file included from cli/main.c:1:
include/compiler.h:78:28: warning: 'OSSpinLock' is deprecated: first deprecated in macOS 10.12 - Use os_unfair_lock() from <os/lock.h> instead [-Wdeprecated-declarations]
    void pthread_spin_init(OSSpinLock * a_lock, void * a_opts);
                           ^
/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/usr/include/libkern/OSSpinLockDeprecated.h:79:17: note: 'OSSpinLock' has been explicitly marked deprecated here
typedef int32_t OSSpinLock OSSPINLOCK_DEPRECATED_REPLACE_WITH(os_unfair_lock);
                ^
2 warnings generated.
make[2]: *** [engine] Error 1
make[1]: *** [mac_osx] Error 2
make: *** [mac_osx] Error 2
```
If someone hasn't already I can start a PR for this.

### PR DESCRIPTION
**Pull request by rjdbcm**
spinlock API updated to fix #7

### CODE REVIEW

File: src/engine/src/_main.c [hunk 1]
```diff
@@ -390,7 +390,7 @@ NO_OPTIMIZATION void destruct_signal_handling(){
 
 
 NO_OPTIMIZATION void set_thread_params(){
-#if SG_OS != _OS_LINUX
+#if SG_OS != _OS_LINUX || SG_OS != _OS_MAC_OS_X
```
Comment (ghost) [inline src/engine/src/_main.c]: I think that should be `&&` instead of `||`, right?

**Comment by ghost** [comment]
Aside from the one comment, this looks great, thanks.  Let me know if that one line is correct or not, then I'll merge this.

BTW, are you using the `make mac_osx` target to build?

**Comment by rjdbcm** [comment]
I did one then the other and yes that should && I was right the first time and then second guessed haha

File: src/engine/src/_main.c (hunk 1 above)
Comment (rjdbcm) [reply src/engine/src/_main.c]: Fixed, Should be ok to merge

**Pull request merged**